forum/qa-include/pages/message.php

```diff
@@ -1,257 +1,288 @@
+                    );
+
+                    qa_send_notification(
+                        $toaccount['userid'],
+                        $toaccount['email'],
+                        $toaccount['handle'],
+                        qa_lang('emails/private_message_subject'),
+                        qa_lang('emails/private_message_body'),
+                        $subs
+                    );
+                }
+                $messagesent = true;
+
+                qa_report_event('u_message', $loginuserid, qa_get_logged_in_handle(), qa_cookie_get(), array(
+                    'userid' => $toaccount['userid'],
+                    'handle' => $toaccount['handle'],
+                    'messageid' => $messageid,
+                    'message' => $inmessage,
+                ));
+
+                // show message as part of general history
+                if (qa_opt('show_message_history'))
+                    qa_redirect(qa_request(), array('state' => ($messagesent ? 'message-sent' : 'email-error')));
+            }
+        }
+    }
+
+
+//    Prepare content for theme
+
+    $hideForm = !empty($pageerror) || $messagesent;
+
+    $qa_content['title'] = qa_lang_html('misc/private_message_title');
+
+    $qa_content['error'] = @$pageerror;
+
+    $blockedPrivateMessages = qa_db_query_sub('SELECT `fromUserId`, `toUserId` FROM ^blockedpw WHERE fromUserId = # AND toUserId = # OR fromUserId = # AND toUserId = #', $loginuserid, $toaccount['userid'], $toaccount['userid'], $loginuserid);
+    $allowedPrivateMessages = qa_opt('allow_private_messages');
+    $blockedPrivateMessageBool = 0 != $blockedPrivateMessages->num_rows;
+
+    if (
+        $blockedPrivateMessageBool ||
+        !$allowedPrivateMessages
+    ) {
+        $userLevel = qa_get_logged_in_level();
+        if($userLevel < QA_USER_LEVEL_EDITOR) {
+            // user blocked pw and loggedin can't send messages
+            $qa_content['custom'] = 'Nie możesz wysłać wiadomości prywatnej do tego uzytkownika.';
+        } else {
+            // user blocked pw, but loggedin is admin and can send message
+            $qa_content['form_message'] = [
+                'tags' => 'method="post" action="' . qa_self_html() . '"',
+                'style' => 'tall',
+                'ok' => $messagesent ? qa_lang_html('misc/message_sent') : null,
+                'fields' => [
+                    'message' => [
+                        'type' => $hideForm ? 'static' : '',
+                        'label' => qa_lang_html_sub('misc/message_for_x', qa_get_one_user_html($handle, false)),
+                        'tags' => 'name="message" id="message"',
+                        'value' => qa_html(@$inmessage, $messagesent),
+                        'rows' => 8,
+                        'note' => qa_lang_html_sub('misc/message_explanation', qa_html(qa_opt('site_title'))),
+                        'error' => qa_html(@$errors['message']),

```

Comment:
@ do wywalenia

Reply:
To jest kod q2a, niezbyt chce go ruszać.